core/unit_test/TestMDRangePolicyConstructors.hpp

```diff
@@ -138,4 +138,43 @@ TEST(TEST_CATEGORY_DEATH, policy_invalid_bounds) {
 }
 #endif
 
+TEST(TEST_CATEGORY, policy_get_tile_size) {
+  constexpr int rank = 3;
+  using Policy    = Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<rank>>;
+  using tile_type = typename Policy::tile_type;
+
```

Comment:
Please assert that the recommended tile size is "valid", that is the tile size is non-negative (if appropriate) for each dimension and the total flattened size does not exceed the limit.

Reply:
Added checks to confirm that the recommended tile size per rank is positive value and is less than the recommended largest tile size per rank.

Reply:
As for checking the total flattened size not exceeding the limit, I was going to add the following unit test:
```
#ifndef KOKKOS_ENABLE_OPENMPTARGET  // FIXME_OPENMPTARGET
TEST(TEST_CATEGORY_DEATH, policy_tile_size_exceeding_limit) {
  using Policy = Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>;
  using tile_type = typename Policy::tile_type;

  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
  ASSERT_DEATH(
      {
        auto max_threads =
            Kokkos::Impl::get_tile_size_properties(TEST_EXECSPACE())
                .max_threads;
        printf("%d\n", max_threads);
        (void)Policy({0, 0, 0}, {100, 100, 100},
                     tile_type{{max_threads, max_threads, max_threads}});
      }, "");
}
#endif
```
but it seems that there may be some deficiencies in `MDRangePolicy` that checks for this condition.

https://github.com/kokkos/kokkos/blob/98b1a38e5391c57b866c4b16bf8108fce56d827a/core/src/KokkosExp_MDRangePolicy.hpp#L379-L387
Currently the product of all tile sizes are checked against `properties.max_threads` instead of `properties.max_total_tile_size`. For Cuda and HIP, `properties.max_total_tile_size != properties.max_threads`, so some inconsistencies are found when running the unit test. I think this part will need to be addressed separately.